src/app/choose-template/page.tsx

```diff
@@ -0,0 +1,60 @@
+'use client' // we need this to be a client component because nhsuk-react-components uses client-only react features
+import { Radios, Button, ErrorSummary, ErrorMessage,  } from 'nhsuk-react-components';
+import { useFormState } from 'react-dom';
+import { chooseTemplateServerAction, FormState } from './choose-template-server-action';
+import { chooseTemplatePageContent } from '../../content/content';
+
+export default function ChooseTemplatePage() {
+    const { pageHeading, errorHeading, options } = chooseTemplatePageContent;
+    const [state, action] = useFormState<FormState, FormData>(chooseTemplateServerAction, { formErrors: [], fieldErrors: {}});
+    console.log(state);
+    console.log(Object.entries(state.fieldErrors));
+    const hasError = state.formErrors.length > 0 || Object.keys(state.fieldErrors).length > 0;
+    return (
+        <>
+            {hasError && (
+            <div>
+                <ErrorSummary>
+                    <ErrorSummary.Title data-testid='error-summary'>
+                        {errorHeading}
+                    </ErrorSummary.Title>
+                    <ErrorSummary.List>
+                        {
+                            Object.entries(state.fieldErrors).map(([id, errors]) => (
+                                <ErrorSummary.Item href='#choose-template' key={`field-error-summary-${id}`}>{errors.join(', ')}</ErrorSummary.Item>
+                            ))
+                        }
+                        {
+                            Object.entries(state.formErrors).map((error, id) => (
+                                <ErrorSummary.Item key={`form-error-summary-${id}`}>{error}</ErrorSummary.Item>
+                            ))
+                        }
+                    </ErrorSummary.List>
+                </ErrorSummary>
+            </div>
+            ) }
+            <form action={action}>
```

Comment:
Potential candidate for a refactor (separate function or non-reusable component in the same directory).